builder/build.go

```diff
@@ -209,9 +209,46 @@ func createBuildTemplate(functionName string, handler string, language string, u
 		}
 	}
 
+	for _, extraPath := range copyExtraPaths {
+		extraPathAbs, err := pathInScope(extraPath, ".")
```

Comment:
After this check, should the same logic be used as the above "Infos" loop in each "extra path" folder to ensure a `build` and `template` folder does not get copied in? I'm not sure if the reasoning for that check above really only comes into play when the function handler is the root of the project or if it matters all the time.

Reply:
i am not sure that check is needed because the use is explicitly adding these extra paths. In the initial infos loop, we are doing implicit copies, so i think it makes sense to skip.

However, we should definitely check that they don't add `.`, I dont think that makes sense, which is exactly your comment below https://github.com/openfaas/faas-cli/pull/744/files/e93706a1ce32103171da538c01b4bf3857fda6d9#r363798976

I will update that and add a test for it